pyomo/contrib/incidence_analysis/interface.py

```diff
@@ -467,38 +640,68 @@ def block_triangularize(self, variables=None, constraints=None):
         sccs = get_scc_of_projection(graph, con_nodes)
         row_idx_map = {r: idx for idx, scc in enumerate(sccs) for r, _ in scc}
         col_idx_map = {c-M: idx for idx, scc in enumerate(sccs) for _, c in scc}
-        # Cache maps in case we want to get diagonal blocks quickly in the
-        # future.
-        row_block_map = row_idx_map
-        col_block_map = {j-M: idx for j, idx in col_idx_map.items()}
-        self.row_block_map = row_block_map
-        self.col_block_map = col_block_map
         con_block_map = ComponentMap(
-            (constraints[i], idx) for i, idx in row_block_map.items()
+            (constraints[i], idx) for i, idx in row_idx_map.items()
         )
         var_block_map = ComponentMap(
-            (variables[j], idx) for j, idx in col_block_map.items()
+            (variables[j], idx) for j, idx in col_idx_map.items()
         )
         # Switch the order of the maps here to match the method call.
         # Hopefully this does not get too confusing...
         return var_block_map, con_block_map
 
-    # TODO: Update this method with a new name and deprecate old name.
-    def get_diagonal_blocks(self, variables=None, constraints=None):
-        """
-        Returns the diagonal blocks in a block triangularization of the
-        incidence matrix of the provided constraints with respect to the
-        provided variables.
+    def block_triangularize(self, variables=None, constraints=None):
+        """Compute an ordered partition of the provided variables and
+        constraints such that their incidence matrix is block lower triangular
+
+        Subsets in the partition correspond to the strongly connected components
+        of the bipartite incidence graph, projected with respect to a perfect
+        matching.
 
         Returns
         -------
-        tuple of lists
-        The first list contains lists that partition the variables,
-        the second lists contains lists that partition the constraints.
+        var_partition: list of lists
+            Partition of variables. The inner lists hold unindexed variables.
+        con_partition: list of lists
+            Partition of constraints. The inner lists hold unindexed constraints.
+
+        Example
+        -------
+        >>> import pyomo.environ as pyo
+        >>> from pyomo.contrib.incidence_analysis import IncidenceGraphInterface
+        >>> m = pyo.ConcreteModel()
+        >>> m.x = pyo.Var([1, 2])
+        >>> m.eq1 = pyo.Constraint(expr=m.x[1]**2 == 7)
+        >>> m.eq2 = pyo.Constraint(expr=m.x[1]*m.x[2] == 3)
+        >>> igraph = IncidenceGraphInterface(m)
+        >>> vblocks, cblocks = igraph.block_triangularize()
+        >>> print([[v.name for v in vb] for vb in vblocks])
+        [['x[1]'], ['x[2]']]
+        >>> print([[c.name for c in cb] for cb in cblocks])
+        [['eq1'], ['eq2']]
```

Comment:
You should add the `.. doctest::` command in this docstring so that this short example is tested (as long as this docstring is included in API documentation on RTD)